src/modules/nfts/components/wizard/Summary.tsx

```diff
@@ -0,0 +1,193 @@
+        </Attribute>
+      )}
+      {value.attributes?.map((attribute: NFTAttribute, index: number) =>
+        attribute.trait_type ? (
+          <Attribute key={index}>
+            <Paragraph style={{ width: 110 }}>{attribute.trait_type}:</Paragraph>
+            <Paragraph>{attribute.value}</Paragraph>
+          </Attribute>
+        ) : null
+      )}
+      {showCreatorCount && (
+        <Attribute>
+          <Paragraph style={{ width: 110 }}>Creators:</Paragraph>
+          <Paragraph>{value.properties.creators.length - 1}</Paragraph>
+        </Attribute>
+      )}
+    </StyledSummaryItem>
+  );
+};
+export default function Summary({
+  previousStep,
+  goToStep,
+  images,
+  nextStep,
+  dispatch,
+  formValues,
+  goToNamedStep,
+  setNFTValues,
+  form,
+}: Props) {
+  const [isUploading, setIsUploading] = useState(false);
+  // TODO: Can extract this to top level component
+  const upload = async () => {
+    const body = new FormData();
+
+    images.forEach((i) => body.append(i.name, i, i.name));
+
+    setIsUploading(true);
+    try {
+      const resp = await fetch('/api/ipfs/upload', {
+        method: 'POST',
+        body,
+      });
+
+      const uploadedFilePins = await resp.json();
+      dispatch({ type: 'UPLOAD_FILES', payload: uploadedFilePins.files });
+      setNFTValues(uploadedFilePins.files);
+      nextStep!();
+    } catch {
+      setIsUploading(false);
+    }
+  };
+
+  if (!formValues) return null;
+
+  return (
+    <NavContainer title="Summary" previousStep={previousStep} goToStep={goToStep}>
+      <Header>Do these look right?</Header>
+      <Button
+        onClick={upload}
+        type="primary"
+        style={{ display: 'flex', alignItems: 'center' }}
+        disabled={isUploading}
+      >
+        Looks good
+        {isUploading && <Spinner style={{ marginLeft: 8, marginTop: 5 }} height={24} width={24} />}
+      </Button>
+
+      <Row style={{ marginTop: 78 }}>
+        <Grid>
+          {formValues.map(
+            (fv: NFTFormValue, index: number) =>
+              images[index] && (
+                <SummaryItem
+                  key={fv.name}
+                  value={fv}
+                  image={images[index]}
+                  showRoyaltyPercentage={formValues.some((nft1) =>
+                    // if one or more NFTs have a different royalty percentage it makes sense to show it in the summary
+                    formValues.some(
+                      (nft2) => nft1.seller_fee_basis_points !== nft2.seller_fee_basis_points
+                    )
+                  )}
+                  showCreatorCount={formValues.some((nft1) =>
+                    // if one or more NFTs have a different number of creators(other than seller and holaplex) it makes sense to show it in the summary.
+                    formValues.some(
+                      (nft2) => nft1.properties.creators.length !== nft2.properties.creators.length
+                    )
+                  )}
```

Comment:
These checks are potentially expensive. can we move them into a `useMemo` hook?

Reply:
Nice! I added them as memo functions like this now:
```
// if one or more NFTs have a different royalty percentage it makes sense to show it in the summary
  const showRoyaltyPercentage = useMemo(() => {
    return (
      formValues?.some((nft1) =>
        formValues.some((nft2) => nft1.seller_fee_basis_points !== nft2.seller_fee_basis_points)
      ) ?? false
    );
  }, [formValues]);

  // if one or more NFTs have a different number of creators(other than seller and holaplex) it makes sense to show it in the summary.
  const showCreatorCount = useMemo(
    () =>
      formValues?.some((nft1) =>
        formValues.some(
          (nft2) => nft1.properties.creators.length !== nft2.properties.creators.length
        )
      ) ?? false,
    [formValues]
  );
```

Reply:
Perfect!